benchmark_data_loading/dataset_helpers.py

```diff
@@ -194,8 +209,20 @@ def with_DL(obj):
             datapipe_adapter_fn=adapter.Shuffle(),
             reading_service=MultiProcessingReadingService(num_workers=args.num_workers),
         )
-
-    if isinstance(obj, ImageFolder):
+    elif isinstance(obj, ImageFolder):
         return data.DataLoader(obj, batch_size=1, collate_fn=lambda x: x, num_workers=args.num_workers, shuffle=True)
+    elif isinstance(obj, wds.WebDataset):
+        return (
+            wds.WebLoader(
+                obj,
+                batch_size=None,
+                num_workers=args.num_workers,
+            )
+            .shuffle(wds_archive_size, initial=wds_archive_size)  # intra-archive shuffling
+            .batched(
+                batchsize=1,
```

Comment:
I changed my mind, let's use `batchsize=batch_size` here as well. I just added it for the other DL in the latest changes https://github.com/NicolasHug/vision/blob/benchmark_data_loading/benchmark_data_loading/dataset_helpers.py#L223